includes/Modules/Idea_Hub.php

```diff
@@ -730,4 +693,89 @@ private function is_idea_post( $post_id ) {
 		return is_array( $this->get_post_idea( $post_id ) );
 	}
 
+	/**
+	 * Gets the parent slug to use for Idea Hub API requests.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string Parent slug.
+	 */
+	private function get_parent_slug() {
+		$reference_url = $this->context->get_reference_site_url();
+		$reference_url = rawurlencode( $reference_url );
+
+		$parent = "platforms/sitekit/properties/{$reference_url}";
+
+		return $parent;
+	}
+
+	/**
+	 * Fetches ideas from the Idea Hub API.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param string $type Ideas type. Valid values "saved", "new" or an empty string which means all ideas.
+	 * @return mixed List ideas request.
+	 */
+	private function fetch_ideas( $type ) {
+		$parent = $this->get_parent_slug();
+		$params = array(
+			'pageSize' => 100,
+		);
+
+		if ( 'saved' === $type ) {
+			$params['filter'] = 'saved(true)';
+		} elseif ( 'new' === $type ) {
+			$params['filter'] = 'saved(false)';
+		}
+
+		$service = $this->get_service( 'ideahub' );
+		$ideas   = $service->platforms_properties_ideas;
+
+		return $ideas->listPlatformsPropertiesIdeas( $parent, $params );
+	}
+
+	/**
+	 * Filters out ideas for which we have already created a post.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $ideas Ideas list to filter.
+	 * @return array Filtered ideas list.
+	 */
+	private function filter_out_drafted_ideas( $ideas ) {
```

Comment:
This method name _right now_ is technically inaccurate, since at the moment it queries post ideas regardless of "draft" post status.
But anyway, see my other comment reply, this isn't exactly what we need to do here. It should filter out ideas for which there's a post of _any_ post status other than "trash".

Reply:
Updated.